### PR TITLE
Add `driver` parameter to `get_uri()` to facilitate psycopg3 support

### PR DESCRIPTION
**Pull request by aaron-siegel**
Also updates the tests to check that SQLAlchemy works with both psycopg2 and psycopg3.

### CODE REVIEW

**Comment by aaron-siegel** [comment]
This change is [Reviewable](https://reviewable.io/reviews/pixeltable/pixeltable-pgserver/10)

File: pyproject.toml [hunk 1]
```diff
@@ -26,11 +26,15 @@ dev = [
 ]
 test = [
     "pytest",
-    "psycopg2-binary",
+    "psycopg2-binary==2.9.9",
```
Comment (mkornacker) [inline pyproject.toml:29]: We still need this?

File: src/pixeltable_pgserver/postgres_server.py [hunk 2]
```diff
@@ -75,10 +76,10 @@ def get_pid(self) -> Optional[int]:
         """
         return self.get_postmaster_info().pid
 
-    def get_uri(self, database : Optional[str] = None) -> str:
+    def get_uri(self, database: Optional[str] = None, driver: Optional[str] = None) -> str:
```
Comment (mkornacker) [inline src/pixeltable_pgserver/postgres_server.py:79]: What's the reason for making the driver configurable?

**Review by aaron-siegel — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pixeltable/pixeltable-pgserver/10)* status: 0 of 4 files reviewed, 2 unresolved discussions

___
*[`pyproject.toml` line 29 at r1](https://reviewable.io/reviews/pixeltable/pixeltable-pgserver/10#gh-1743018700:-O5yV3dM4BLROZXD3mvM:bavli6j) ([raw file](https://github.com/pixeltable/pixeltable-pgserver/blob/911d0b9df6be1c46b6aec4ed6a1d14419b59ca1b/pyproject.toml#L29)):*
<details><summary><i>Previously, mkornacker (Marcel Kornacker) wrote…</i></summary><blockquote>

We still need this?
</blockquote></details>

It's the easiest thing to do to facilitate the migration and keep our options open, at least until we're 100% comfortable with psycopg3.
___
*[`src/pixeltable_pgserver/postgres_server.py` line 79 at r1](https://reviewable.io/reviews/pixeltable/pixeltable-pgserver/10#gh-1743028722:-O5yVCo58BYc4bF1A_KR:b-2lov1a) ([raw file](https://github.com/pixeltable/pixeltable-pgserver/blob/911d0b9df6be1c46b6aec4ed6a1d14419b59ca1b/src/pixeltable_pgserver/postgres_server.py#L79)):*
<details><summary><i>Previously, mkornacker (Marcel Kornacker) wrote…</i></summary><blockquote>

What's the reason for making the driver configurable?
</blockquote></details>

SQLAlchemy needs the driver in the URL, and direct Postgres needs the URL without driver. So at minimum, we need `None` and `'psycopg'` variants. This seems like the most flexible/logical way to handle that

**Pull request merged**